static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+import config from 'lib/config';
+import { local } from 'lib/storage';
+import template from 'lodash/utilities/template';
+import Message from 'common/modules/ui/message';
+import messageTemplate from 'raw-loader!common/views/membership-message.html';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import mediator from 'lib/mediator';
+import { testCanBeRun } from 'common/modules/experiments/test-can-run-checks';
+import MembershipEngagementBannerTests from 'common/modules/experiments/tests/membership-engagement-banner-tests';
+import { inlineSvg } from 'common/views/svgs';
+import { isInTest, variantFor } from 'common/modules/experiments/segment-util';
+import { epicEngagementBannerTests } from 'common/modules/experiments/acquisition-test-selector';
+import membershipEngagementBannerUtils from 'common/modules/commercial/membership-engagement-banner-parameters';
+import { isBlocked } from 'common/modules/commercial/membership-engagement-banner-block';
+import ophan from 'ophan/ng';
+import { get } from 'lib/geolocation';
+import { constructQuery } from 'lib/url';
+
+// change messageCode to force redisplay of the message to users who already closed it.
+const messageCode = 'engagement-banner-2017-07-05';
+
+const DO_NOT_RENDER_ENGAGEMENT_BANNER = 'do no render engagement banner';
+
+const getUserTest = (): ?ContributionsABTest => {
+    const engagementBannerTests = MembershipEngagementBannerTests.concat(
+        epicEngagementBannerTests
+    );
+
+    return engagementBannerTests.find(
+        test => testCanBeRun(test) && isInTest(test)
+    );
+};
+
+const getUserVariant = (test: ?ABTest): ?Variant =>
+    test ? variantFor(test) : undefined;
+
+const buildCampaignCode = (
+    offering: string,
+    campaignId: string,
+    variantId: string
+): string => {
+    let prefix = '';
+    const offerings = membershipEngagementBannerUtils.offerings;
+
+    // mem and cont chosen to be consistent with default campaign code prefixes.
+    if (offering === offerings.membership) {
+        prefix = 'mem_';
+    } else if (offering === offerings.contributions) {
+        prefix = 'cont_';
+    }
+
+    return `${prefix + campaignId}_${variantId}`;
+};
+
+const getUserVariantParams = (
+    userVariant: ?Variant,
+    campaignId: ?string,
+    defaultOffering: string
+): Object => {
+    if (campaignId && userVariant && userVariant.engagementBannerParams) {
```

Comment:
I'm checking for the existence of `campaignId` here to avoid building a meaningless campaign code